GitUI/SpellChecker/EditNetSpell.cs

```diff
@@ -870,16 +870,32 @@ private string GetWordAtCursor()
             if (TextBox.Text.Length > TextBox.SelectionStart && IsSeparatorChar(TextBox.Text[TextBox.SelectionStart]))
                 return null;
 
+            return ExtractWord(TextBox.Text, TextBox.SelectionStart - 1);
+        }
+
+        private string ExtractWord(string text, int cursor)
+        {
             var sb = new StringBuilder();
 
-            int i = TextBox.SelectionStart - 1;
+            while (cursor >= 0)
+            {
+                if (text[cursor] == ' ')
+                    break;
 
-            while (i >= 0 && !IsSeparatorChar(TextBox.Text[i]))
-                sb.Insert(0, TextBox.Text[i--]);
+                if (text[cursor] == '.' && !IsWordLeadingChar(text, cursor))
+                    break;
+
+                sb.Insert(0, text[cursor--]);
+            }
 
             return sb.ToString();
         }
 
+        private bool IsWordLeadingChar(string text, int cursor)
+        {
+            return cursor == 0 || text[cursor - 1] == ' ';
```

Comment:
your implementation assumes space as a word delimiter
this is only partially correct - there can be other delimiters, e.g. tab, newline, various braces etc